src/main/java/org/saltyrtc/client/helpers/ValidationHelper.java

```diff
@@ -56,18 +56,28 @@ public static String validateType(Object value, String expected) throws Validati
      * so an array is always deserialized as Object[].
      */
     @SuppressWarnings("unchecked")
-    public static <T> List<T> validateTypedList(Object values, Class type, String name) throws ValidationError {
+    public static <T> List<T> validateTypedList(Object values, Class type, String name, boolean allowNull) throws ValidationError {
         if (!(values instanceof List)) {
             throw new ValidationError(name + " must be a list");
         }
         for (Object element : (List) values) {
-            if (!type.isInstance(element)) {
+            if ((element == null && !allowNull) || !type.isInstance(element)) {
                 throw new ValidationError(name + " must be a " + type.getSimpleName() + " list");
```

Comment:
`type.getSimpleName()` will fail if `allowNull` is `false` and `element` is null